src/plugins/intel_gpu/src/plugin/transformations/unsqueeze_broadcast_reshape_sdpa_fusion.cpp

```diff
@@ -63,26 +73,52 @@ UnsqueezeBroadcastReshapeSDPAFusion::UnsqueezeBroadcastReshapeSDPAFusion() {
         }
         const auto& pattern_map = m.get_pattern_value_map();
 
-        auto valid_broadcast_target_shape = [](const std::vector<int32_t>& target_shape) {
-            return std::count_if(target_shape.begin(), target_shape.end(), [](int32_t s) { return s != 1; }) == 1;
-        };
+        bool has_pre_reshape = (pattern_map.find(pre_reshape_b_m) != pattern_map.end()) &&
+                               (pattern_map.find(pre_reshape_c_m) != pattern_map.end());
+
         auto broadcast_b = ov::as_type_ptr<ov::op::v3::Broadcast>(pattern_map.at(broadcast_b_m).get_node_shared_ptr());
         auto broadcast_c = ov::as_type_ptr<ov::op::v3::Broadcast>(pattern_map.at(broadcast_c_m).get_node_shared_ptr());
 
+        auto get_input_shape = [](const std::shared_ptr<ov::op::v3::Broadcast>& broadcast) -> std::vector<int32_t> {
+            if (!broadcast) return {};
+            auto input_node = broadcast->get_input_node_shared_ptr(0);
+            if (input_node && input_node->get_output_partial_shape(0).is_static()) {
+                auto pshape = input_node->get_output_shape(0);
+                return std::vector<int32_t>(pshape.begin(), pshape.end());
+            }
+            return {};
+        };
+
+        auto valid_broadcast_target_shape = [has_pre_reshape](const std::vector<int32_t>& input_shape,
+                                                              const std::vector<int32_t>& target_shape) {
+            if (has_pre_reshape) {
+                if (input_shape.empty() || (input_shape.size() != target_shape.size())) return false;
```

Comment:
This change allows only static input If there is a reshape before broadcast.
shouldn't the logic be 
auto valid_broadcast_target_shape : 
   if dst_shape_is_static : 
          count_diff_of_broadcast_source_and_target_shape <= 1;
   else 
         return original condition

Reply:
updated as suggested. but count_diff_of_broadcast_source_and_target_shape should still be 1, even if it's static